UKHouses/__init__.py

```diff
@@ -32,74 +31,73 @@
 pagination = Pagination(app, db)
 
 
-##########################################
-### 2. create a model from existing DB ###
-##########################################
-
-# MODELS
+# MODELS -  a helper that help us ti retrieve HousePricesModel objects from the db - internal repr of an Entity
 class HousePricesModel(db.Model):
     __table__ = db.Model.metadata.tables['HousePrices']
     __table_args__ = {'autoload': True}
 
-    # Since I am using reflection I don't need a constructor. Define what to return.
     def json(self):
         return {'Id': self.Id, 'code': self.Code, 'price': self.price,
-                'sold date': self.date.isoformat(sep=' '), 'postcode': self.postcode, 'Property Type': self.propType,
+                'date': (self.date).isoformat(), 'postcode': self.postcode, 'Property Type': self.propType,
                 'New built?': self.newBuild, 'Estate Type': self.estateType, 'House/flat number': self.number,
                 'Street Address': self.street, 'town': self.town, 'district': self.district, 'county': self.county}
 
-    #################################################
-    #### 3. perform CRUD to later move to crud.py ###
-    #################################################
-
 
 """
 Resource fields for marshalling
 """
 page_fields = {
-    'Id': fields.String, 'code': fields.String, 'price': fields.String, 'sold date': fields.String,
+    'Id': fields.String, 'code': fields.String, 'price': fields.String, 'date': fields.String,
     'postcode': fields.String, 'Property Type': fields.String, 'New built?': fields.String,
     'Estate Type': fields.String, 'House/flat number': fields.String, 'Street Address': fields.String,
     'town': fields.String, 'district': fields.String, 'county': fields.String
 }
 
 
-# RESOURCES
+# RESOURCES - external repr of an Entity, it maps endpoints
+class Home(Resource):
+    def get(self):
+        return render_template('index.html'), 200
+
+
 class House(Resource):
     def get(self, Id):
-        tw = HousePricesModel.query.filter_by(Id=Id).first()
-        if tw:
-            return tw.json()
+        houseId = HousePricesModel.query.filter_by(Id=Id).first()
+        if houseId:
+            return {"house": houseId.json()}
         else:
             return {'Id': None}, 404
 
 
 class HouseList(Resource):
     def get(self):
-        houseprices = HousePricesModel.query.all()
-        return pagination.paginate([hp.json() for hp in houseprices], page_fields)
-
+        # print(request.args)
+        args = request.args
+        from_date = args.get('from')
+        to_date = args.get("to")
+        if request.method == 'GET' and 'from' in args is not None:
```

Comment:
Consider checking for `to` in the args as well:
`if request.method == 'GET' and 'from' in args and 'to' in args:`.
That we we can be sure both args are provided :)